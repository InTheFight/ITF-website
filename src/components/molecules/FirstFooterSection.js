import React from 'react';
import PropTypes from 'prop-types';

import styled from 'styled-components';

const Container = styled.div`
  height: 460px;
  background: #2c358f;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  padding-left: 150px;
  padding-right: 150px;
`;

const Title = styled.h2`
  font-family: 'Gelo';
  font-size: 32px;
  text-align: center;
  color: #faf5eb;
`;

const Text = styled.p`
  font-family: 'Gelo';
  font-size: 18px;
  text-align: center;
  color: #faf5eb;
`;

const Button = styled.button`
  background: #faf5eb;
  color: #2c358f;
  padding: 15px 40px 15px 40px;
  font-weight: bold;
`;

const FirstFooterSection = ({ title, text, buttonText, buttonAction }) => (
  <Container>
    <Title>{title}</Title>
    <Text>{text}</Text>
    <Button onClick={buttonAction}>{buttonText}</Button>
  </Container>
);

FirstFooterSection.propTypes = {
  title: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  buttonText: PropTypes.string.isRequired,
  buttonAction: PropTypes.func.isRequired,
};

export default FirstFooterSection;
