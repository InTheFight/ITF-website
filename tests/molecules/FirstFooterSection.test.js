import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import FirstFooterSection from '../../src/components/molecules/FirstFooterSection';

test('Displays the subscription form', () => {
  const footerSection = render(<FirstFooterSection title='Subscribe to get updates!' />);

  expect(screen.getByRole('heading')).toHaveTextContent('Subscribe to get updates!');
  expect(footerSection).toMatchSnapshot();
});
