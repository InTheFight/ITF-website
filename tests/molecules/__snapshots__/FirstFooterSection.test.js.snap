// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays the subscription form 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnylx ByCpw"
      >
        <h2
          class="sc-gtssRu iQQAXn"
        >
          Subscribe to get updates!
        </h2>
        <form
          class="sc-hKFyIo korkGl"
        >
          <label
            class="sc-dlnjPT iyhkgJ"
          >
            Name:
            <input
              class="sc-eCApGN bWoYLv"
              name="name"
              type="text"
            />
          </label>
          <label
            class="sc-dlnjPT iyhkgJ"
          >
            Email:
            <input
              class="sc-eCApGN bWoYLv"
              name="email"
              type="text"
            />
          </label>
          <input
            class="sc-jSFkmK eicLKk"
            type="submit"
            value="Submit"
          />
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnylx ByCpw"
    >
      <h2
        class="sc-gtssRu iQQAXn"
      >
        Subscribe to get updates!
      </h2>
      <form
        class="sc-hKFyIo korkGl"
      >
        <label
          class="sc-dlnjPT iyhkgJ"
        >
          Name:
          <input
            class="sc-eCApGN bWoYLv"
            name="name"
            type="text"
          />
        </label>
        <label
          class="sc-dlnjPT iyhkgJ"
        >
          Email:
          <input
            class="sc-eCApGN bWoYLv"
            name="email"
            type="text"
          />
        </label>
        <input
          class="sc-jSFkmK eicLKk"
          type="submit"
          value="Submit"
        />
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
